types/inferschematype.d.ts

```diff
@@ -160,7 +159,7 @@ type ResolvePathType<PathValueType, Options extends SchemaTypeOptions<PathValueT
           PathValueType extends DateSchemaDefinition ? Date :
             PathValueType extends typeof Buffer | 'buffer' | 'Buffer' | typeof Schema.Types.Buffer ? Buffer :
               PathValueType extends BooleanSchemaDefinition ? boolean :
-                PathValueType extends ObjectIdSchemaDefinition ? Types.ObjectId :
+                PathValueType extends 'objectId' | 'ObjectId' | typeof Schema.Types.ObjectId ? Types.ObjectId :
```

Comment:
Why not just add 'objectid' to `ObjectIdSchemaDefinition` ?

Reply:
@vkarpov15 
Which values string values that should be resolved to object id ?

Reply:
@iammola 
You might need to as well to make this change as well:
```diff
 - PathValueType extends Record<string, any> ? ObtainDocumentType<PathValueType, any, TypeKey> :
+ PathValueType extends Record<string, any> ? IfEquals<PathValueType, Types.ObjectId, Types.ObjectId, ObtainDocumentType<PathValueType, any, TypeKey>> :

```

Reply:
@vkarpov15  Please tell me when you feel that we can't improve the performance because of auto typed schema, so I can the entire feature, I wouldn't like to make mongoose worse.

Reply:
I ran benchmark test after deleting the entire types folder, I got this result:
We just might consider that, I think.
![Capture](https://user-images.githubusercontent.com/50488054/182685997-64603e0f-974f-4b4e-842c-92dbe485aef9.PNG)

Reply:
Deleting all the types is a quite radical approach, imho

Reply:
I was just wondering what I will get if it has no types.

Reply:
I think it is weird to get this result even if the package have no type definition files. 
```
Instantiations               15723
Memory⠀used             129661K
Check⠀time                  1.36s
Total⠀time                    2.31s
```

Reply:
I guess tsc loads also typings from node_modules and maybe global Modules

Reply:
Sure, but  It might be more helpful to be able as well to diagnose only specific package type definitions.